test/e2e/advanced_test.go

```diff
@@ -0,0 +1,67 @@
+package e2e_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions"
+	kubecli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/kubecli"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/model"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/utils"
+)
+
+var _ = Describe("Deploy cluster", Label("cluster-advanced-ns"), func() {
+	var data model.TestDataProvider
+
+	BeforeEach(func() {
+		Eventually(kubecli.GetVersionOutput()).Should(Say(K8sVersion))
+	})
+	AfterEach(func() {
+		GinkgoWriter.Write([]byte("\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		GinkgoWriter.Write([]byte("Operator namespace: " + data.Resources.Namespace + "\n"))
+		GinkgoWriter.Write([]byte("===============================================\n"))
+		if CurrentSpecReport().Failed() {
+			GinkgoWriter.Write([]byte("Test has been failed. Trying to save logs...\n"))
+			utils.SaveToFile(
+				fmt.Sprintf("output/%s/operatorDecribe.txt", data.Resources.Namespace),
+				[]byte(kubecli.DescribeOperatorPod(data.Resources.Namespace)),
+			)
+			utils.SaveToFile(
+				fmt.Sprintf("output/%s/operator-logs.txt", data.Resources.Namespace),
+				kubecli.GetManagerLogs(data.Resources.Namespace),
+			)
+			actions.SaveTestAppLogs(data.Resources)
+			actions.SaveK8sResources(
+				[]string{"deploy", "atlasclusters", "atlasprojects"},
+				data.Resources.Namespace,
+			)
+			actions.AfterEachFinalCleanup([]model.TestDataProvider{data})
+		}
+	})
+
+	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
+		func(test model.TestDataProvider) {
+			data = test
+			mainCycle(test)
+		},
+		Entry("Trial - Simplest configuration with no backup", Label("ns-crd"),
+			model.NewTestDataProvider(
+				"operator-ns-crd",
+				model.AProject{},
+				model.NewEmptyAtlasKeyType().UseDefaulFullAccess(),
+				[]string{"data/atlascluster_basic_with_keep_resource_policy.yaml"},
+				[]string{},
+				[]model.DBUser{},
```

Comment:
Do we need the DBUser?

Reply:
We need to pass `[]DBUser` to `NewTestDataProvider`.

https://github.com/commercetools/mongodb-atlas-kubernetes/blob/c31c99d17ebe4e2df84721e3caa5150e505e6906/test/e2e/model/dataprovider.go#L13

But the decision on the creation of a user is happening in line `437`. so if we don't want to create a user, we need to duplicate the `DeployUserResourcesAction` and ` DeleteUserResources` and remove the user creation|deletion part, which I personally prefer to avoid duplication and wait a little more for the creation of a user!

https://github.com/commercetools/mongodb-atlas-kubernetes/blob/c31c99d17ebe4e2df84721e3caa5150e505e6906/test/e2e/actions/steps.go#L434-L438